addons/web_editor/static/src/js/editor/odoo-editor/test/spec/editor.test.js

```diff
@@ -3839,6 +3839,88 @@ X[]
                     contentAfter: '<p><br></p><p>[]<br></p>',
                 });
             });
+            it('preserve selected table row and insert paragraph in cell when whole table is not selected and enter key is pressed', async () => {
+                // Forward selection
+                await testEditor(BasicEditor, {
+                    contentBefore: unformat(
+                        `<table>
+                            <tbody>
+                                <tr><td><p>[<br></p></td><td><p><br></p></td><td><p><br>]</p></td></tr>
```

Comment:
If you're going to use `unformat` anyway, you might as well format your HTML so it doesn't have a 103-character line that is hard to parse at a glance. One line per `td` would be good.